spec.html

```diff
@@ -32634,34 +32391,31 @@ <h1>Runtime Semantics: CharacterSetMatcher ( _A_, _invert_, _direction_ )</h1>
               1. Let _index_ be min(_e_, _f_).
               1. Let _ch_ be the character _Input_[_index_].
               1. Let _cc_ be Canonicalize(_ch_).
-              1. If _invert_ is *false*, then
-                1. If there does not exist a member _a_ of set _A_ such that Canonicalize(_a_) is _cc_, return ~failure~.
-              1. Else,
-                1. Assert: _invert_ is *true*.
-                1. If there exists a member _a_ of set _A_ such that Canonicalize(_a_) is _cc_, return ~failure~.
+              1. If there exists a member _a_ of _A_ such that Canonicalize(_a_) is _cc_, let _found_ be *true*. Otherwise, let _found_ be *false*.
+              1. If _invert_ is *false* and _found_ is *false*, return ~failure~.
+              1. If _invert_ is *true* and _found_ is *true*, return ~failure~.
               1. Let _cap_ be _x_'s _captures_ List.
               1. Let _y_ be the State (_f_, _cap_).
-              1. Call _c_(_y_) and return its result.
+              1. Return _c_(_y_).
           </emu-alg>
         </emu-clause>
 
         <emu-clause id="sec-runtime-semantics-canonicalize-ch" aoid="Canonicalize">
           <h1>Runtime Semantics: Canonicalize ( _ch_ )</h1>
           <p>The abstract operation Canonicalize takes argument _ch_ (a character). It performs the following steps when called:</p>
           <emu-alg>
-            1. If _IgnoreCase_ is *false*, return _ch_.
-            1. If _Unicode_ is *true*, then
+            1. If _Unicode_ is *true* and _IgnoreCase_ is *true*, then
               1. If the file CaseFolding.txt of the Unicode Character Database provides a simple or common case folding mapping for _ch_, return the result of applying that mapping to _ch_.
               1. Return _ch_.
-            1. Else,
-              1. Assert: _ch_ is a UTF-16 code unit.
-              1. Let _s_ be the String value consisting of the single code unit _ch_.
-              1. Let _u_ be the same result produced as if by performing the algorithm for `String.prototype.toUpperCase` using _s_ as the *this* value.
-              1. Assert: Type(_u_) is String.
-              1. If _u_ does not consist of a single code unit, return _ch_.
-              1. Let _cu_ be _u_'s single code unit element.
-              1. If the numeric value of _ch_ &ge; 128 and the numeric value of _cu_ &lt; 128, return _ch_.
-              1. Return _cu_.
+            1. If _IgnoreCase_ is *false*, return _ch_.
```

Comment:
Why was this branch moved down?

Reply:
We suspect that this is to call out that the first branch only applies when both _Unicode_ and _IgnoreCase_ is *true*, which would be a net readability improvement, so I'm fine with this.